src/modules/playback-engine/output/chromecast/chromecast-output.ts

```diff
@@ -0,0 +1,278 @@
+    logger.warn(NOT_IMPLEMENTED);
+  }
+
+  setPreload() {
+    logger.warn(NOT_IMPLEMENTED);
+  }
+
+  setSrc(source: PlayableMediaSource, startTime?: number) {
+    if (!source) {
+      return;
+    }
+    let _source = source;
+    let mimeType, url;
+
+    if (Array.isArray(source)) {
+      _source = source[0];
+    }
+
+    if (typeof _source === 'string') {
+      url = _source;
+      mimeType = getMimeByUrl(_source);
+    } else {
+      _source = _source as IPlayableSource;
+      mimeType = getMimeByType(_source.type);
+      url = _source.url;
+    }
+
+    this._src = url;
+
+    const session = cast.framework.CastContext.getInstance().getCurrentSession();
+    const mediaInfo = new chrome.cast.media.MediaInfo(url, mimeType);
+    const request = new chrome.cast.media.LoadRequest(mediaInfo);
+
+    if (startTime) {
+      request.currentTime = startTime;
+    }
+
+    this._stateEngine.setState(EngineState.SRC_SET);
+
+    session.loadMedia(request).then(() => {
+      this._initRemote();
+      this._stateEngine.setState(EngineState.PLAYING);
+    });
+  }
+
+  private _initRemote() {
+    this._player = new cast.framework.RemotePlayer();
+    this._playerController = new cast.framework.RemotePlayerController(
+      this._player,
+    );
+
+    this._stateEngine = new StateEngine(
+      this._eventEmitter,
+      this,
+      this._playerController,
+    );
+  }
+
+  // @ts-ignore
+  getElement() {
+    logger.warn(NOT_IMPLEMENTED);
+    return null;
+  }
+
+  getDebugInfo() {
+    return {
+      output: 'chromecast',
+      currentTime: this.currentTime,
+      duration: this.duration,
+      src: this.src,
+    } as ICromecastDebugInfo;
+  }
+
+  get volume() {
+    return this._player.volumeLevel;
+  }
+
+  get currentTime() {
+    return this._player.currentTime;
+  }
+
+  get duration() {
+    return this._player.duration;
+  }
+
+  get autoplay() {
+    return this._isAutoplay;
+  }
+
+  get crossOrigin() {
+    logger.warn(NOT_IMPLEMENTED);
+    return 'anonymous' as CrossOriginValue;
+  }
+
+  get playbackRate() {
+    return 1;
+  }
+
+  get buffered() {
+    return {
+      end: () => 0,
+      start: () => 0,
+      length: 0,
+    };
+  }
+
+  get preload() {
+    return PreloadType.AUTO;
+  }
+
+  get isPaused() {
+    return this._player.isPaused;
+  }
+
+  get isMuted() {
+    return this._player.isMuted;
+  }
+
+  get isEnded() {
+    return this._stateEngine.state === EngineState.ENDED;
+  }
+
+  get isInline() {
+    return false;
+  }
+
+  get isAutoplay() {
+    return this._isAutoplay;
+  }
+
+  get isSeekAvailable() {
+    return this._player.canSeek;
+  }
+
+  get isLoop() {
+    return false;
+  }
+
+  get isPreloadActive() {
+    return false;
+  }
+
+  get videoHeight() {
+    return 0;
+  }
+
+  get videoWidth() {
+    return 0;
+  }
+
+  get src() {
+    return this._src;
+  }
+
+  get currentState(): EngineState {
+    return this._stateEngine.state;
+  }
+
+  destroy() {
```

Comment:
Do you really need to implement empty destroy (I don't know)? 
Why destroy in the middle of getters?)

Reply:
I need to implement destroy for any module. Since this is not module (anymore), will remove this method.